controllers/addon/monitoring_federation_reconciler_test.go

```diff
@@ -36,53 +41,279 @@ func TestEnsureMonitoringFederation_MonitoringFullyMissingInSpec_NotPresentInClu
 	c.AssertExpectations(t)
 }
 
-func TestEnsureMonitoringFederation_MonitoringPresentInSpec_NotPresentInCluster(t *testing.T) {
+// desired secret in Addon monitoring NS not being present
+func TestEnsureMonitoringFederation_MonitoringPresentInSpec_SecretNotPresent_NotPresentInCluster(t *testing.T) {
 	c := testutil.NewClient()
+	uncachedC := testutil.NewClient()
+	r := &monitoringFederationReconciler{
+		client:                 c,
+		uncachedClient:         uncachedC,
+		scheme:                 testutil.NewTestSchemeWithAddonsv1alpha1(),
+		addonOperatorNamespace: "xxx-addon-operator",
+	}
+
+	addon := testutil.NewTestAddonWithMonitoringFederation()
+	addon.Spec.Monitoring.Federation.PortName = "https"
+
+	c.On("Get", testutil.IsContext, mock.IsType(types.NamespacedName{}), mock.IsType(&corev1.Namespace{}), mock.Anything).
+		Return(testutil.NewTestErrNotFound())
+
+	c.On("Create", testutil.IsContext, mock.IsType(&corev1.Namespace{}), mock.Anything).
+		Run(func(args mock.Arguments) {
+			// mocked Namespace is immediately active
+			namespace := args.Get(1).(*corev1.Namespace)
+			namespace.Status.Phase = corev1.NamespaceActive
+			assert.Equal(t, GetMonitoringNamespaceName(addon), namespace.Name)
+
+		}).
+		Return(nil)
+
+	secretADOtoken := corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "addon-operator-prom-token",
+			Namespace: r.addonOperatorNamespace},
+		Data: map[string][]byte{
+			"token": []byte("mock-token"),
+		},
+	}
+	addonADOtoken := corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{Name: fmt.Sprintf("%s-bearertoken-secret", addon.Name), Namespace: GetMonitoringNamespaceName(addon)},
+		Data: map[string][]byte{
+			"token": []byte("mock-token"),
+		},
+	}
+
+	uncachedC.
+		On("Get",
+			testutil.IsContext,
+			types.NamespacedName{
+				Name:      "addon-operator-prom-token",
+				Namespace: r.addonOperatorNamespace,
+			},
+			mock.IsType(&corev1.Secret{}), mock.Anything).Run(func(args mock.Arguments) {
+		secretADOtoken.DeepCopyInto(args.Get(2).(*corev1.Secret))
+		secret := args.Get(2).(*corev1.Secret)
+		err := controllerutil.SetControllerReference(addon, secret, r.scheme)
+		controllers.AddCommonLabels(secret, addon)
+		assert.NoError(t, err)
+	}).
+		Return(nil)
+	uncachedC.
+		On("Get",
+			testutil.IsContext,
+			types.NamespacedName{
+				Name:      fmt.Sprintf("%s-bearertoken-secret", addon.Name),
+				Namespace: GetMonitoringNamespaceName(addon),
+			},
+			mock.IsType(&corev1.Secret{}), mock.Anything).Run(func(args mock.Arguments) {
+		addonADOtoken.DeepCopyInto(args.Get(2).(*corev1.Secret))
+		secret := args.Get(2).(*corev1.Secret)
+		err := controllerutil.SetControllerReference(addon, secret, r.scheme)
+		controllers.AddCommonLabels(secret, addon)
+		assert.NoError(t, err)
+
```

Comment:
This is not necessary as you are anyways returns a 404. Get rid of the run block.